### PR TITLE
Rem 237

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
refactoring from CustomComboBox to ResourceComboBox

### CODE REVIEW

**Review by NoraBitarSM — CHANGES_REQUESTED** [review]

File: src/pages/activities/forms/ActivityForm.js [hunk 1]
```diff
@@ -45,7 +44,7 @@ export default function ActivityForm({ labels, maxAccess, recordId }) {
     validationSchema: yup.object({
       name: yup.string().required('This field is required'),
       reference: yup.string().required('This field is required'),
-      industry:yup.string().required('This field is required')
+      industry: yup.string().required('This field is required')
```
Comment (omarZrayka23) [inline src/pages/activities/forms/ActivityForm.js:47]: instead This field is required  put ' '

File: src/pages/mc-defaults/index.js [hunk 2]
```diff
@@ -24,185 +23,173 @@ import { MultiCurrencyRepository } from 'src/repositories/MultiCurrencyRepositor
...
+            <ResourceComboBox
+              endpointId={MultiCurrencyRepository.RateType.qry}
+              name='mc_defaultRTFI'
+              label={_labels.mc_defaultRTFI}
+              valueField='recordId'
+              displayField='name'
+              values={formik.values}
+              onChange={(event, newValue) => {
+                if (newValue) {
+                  formik.setFieldValue('mc_defaultRTFI', newValue?.recordId)
```
Comment (omarZrayka23) [inline src/pages/mc-defaults/index.js:169]: you can replace   formik.setFieldValue('mc_defaultRTFI', newValue?.recordId || '')

File: src/pages/mc-defaults/index.js [hunk 3]
```diff
@@ -24,185 +23,173 @@ import { MultiCurrencyRepository } from 'src/repositories/MultiCurrencyRepositor
...
+            <ResourceComboBox
+              endpointId={MultiCurrencyRepository.RateType.qry}
+              name='mc_defaultRTMF'
+              label={_labels.mc_defaultRTMF}
+              valueField='recordId'
+              displayField='name'
+              values={formik.values}
+              onChange={(event, newValue) => {
+                if (newValue) {
+                  formik.setFieldValue('mc_defaultRTMF', newValue?.recordId)
```
Comment (omarZrayka23) [inline src/pages/mc-defaults/index.js:151]: can do  clear if and put ||

File: src/pages/mc-defaults/index.js [hunk 4]
```diff
@@ -24,185 +23,173 @@ import { MultiCurrencyRepository } from 'src/repositories/MultiCurrencyRepositor
...
+          <Grid item xs={12}>
+            <ResourceComboBox
+              endpointId={MultiCurrencyRepository.RateType.qry}
+              name='mc_defaultRTPU'
+              label={_labels.mc_defaultRTPU}
+              valueField='recordId'
+              displayField='name'
+              values={formik.values}
+              onChange={(event, newValue) => {
+                if (newValue) {
```
Comment (omarZrayka23) [inline src/pages/mc-defaults/index.js:132]: can remove if and put ||

File: src/pages/mc-defaults/index.js [hunk 5]
```diff
@@ -24,185 +23,173 @@ import { MultiCurrencyRepository } from 'src/repositories/MultiCurrencyRepositor
...
+          <Grid item xs={12}>
+            <ResourceComboBox
+              endpointId={MultiCurrencyRepository.RateType.qry}
+              name='mc_defaultRTSA'
+              label={_labels.mc_defaultRTSA}
+              valueField='recordId'
+              displayField='name'
+              values={formik.values}
+              onChange={(event, newValue) => {
+                if (newValue) {
```
Comment (omarZrayka23) [inline src/pages/mc-defaults/index.js:114]: can remove if and put ||

File: src/pages/rt-default/index.js [hunk 6]
```diff
@@ -29,7 +28,6 @@ const DocumentTypeMaps = () => {
   //control
   const [labels, setLabels] = useState(null)
   const [access, setAccess] = useState(null)
-  const [numberRangeStore, setNumberRangeStore] = useState([])
 
   //stores
```
Comment (omarZrayka23) [inline src/pages/rt-default/index.js:32]: clear

File: src/pages/rt-default/index.js [hunk 7]
```diff
@@ -82,16 +78,13 @@ const DocumentTypeMaps = () => {
       parameters: parameters
     })
       .then(res => {
-        // console.log(res)
         rtDefaultValidation.setFieldValue('rt-nra-product', res.record.recordId)
```
Comment (omarZrayka23) [inline src/pages/rt-default/index.js:81]: can remove it?

File: src/pages/rt-default/index.js [hunk 8]
```diff
@@ -134,27 +127,13 @@ const DocumentTypeMaps = () => {
...
-      extension: SystemRepository.NumberRange.snapshot,
-      parameters: parameters
-    })
-      .then(res => {
-        setNumberRangeStore(res.list)
-      })
-      .catch()
-  }
-
   console.log('rtDefaultFormValidation', rtDefaultFormValidation)
```
Comment (omarZrayka23) [inline src/pages/rt-default/index.js:137]: remove it

**Pull request merged**